api/v1alpha2/conversion.go

```diff
@@ -155,6 +155,10 @@ func (dst *MachineDeploymentList) ConvertFrom(srcRaw conversion.Hub) error {
 }
 
 func Convert_v1alpha2_MachineSpec_To_v1alpha3_MachineSpec(in *MachineSpec, out *v1alpha3.MachineSpec, s apiconversion.Scope) error {
+	if out.Bootstrap.ConfigRef.Kind == "KubeadmConfig" || out.Bootstrap.ConfigRef.Kind == "KubeadmConfigTemplate" {
```

Comment:
I would also confirm the .Group matches `bootstrap.kubeadm.api.v1alpha2.GroupVersion.Group` (you can alias the package import to `bootstrapv1`).

Reply:
@ncdc for matching Group, this is what I have done-

Reply:
I meant this:

```go
inGV, err := schema.ParseGroupVersion(in.Boostrap.ConfigRef.APIVersion)
if err != nil {
  return err
}
if inGV.Group == bootstrapv1.GroupVersion.Group && (in.Bootstrap.ConfigRef.Kind == "KubeadmConfig" || in.Bootstrap.ConfigRef.Kind == "KubeadmConfigTemplate") {
  out.Bootstrap.ConfigRef.APIVersion = bootstrapv1.GroupVersion.String()
}
// ...
```

Reply:
Get it. However, `bootstrapv1.GroupVersion.String()` would return `v1alpha2` for the `APIVersion` and not `v1alpha3`, isn't it?

Reply:
It's fine to import the v1alpha3 package as `bootstrapv1` - both v1alpha2 and v1alpha3 have the same API group.

Reply:
Thanks. Updated the PR with the changes.